src/OrleansProviders/Streams/Generator/GeneratorPooledCache.cs

```diff
@@ -99,7 +153,10 @@ private ArraySegment<byte> SerializeMessageIntoPooledSegment(GeneratedBatchConta
                 {
                     // no block or block full, get new block and try again
                     currentBuffer = bufferPool.Allocate();
-                    currentBuffer.SetPurgeAction(PurgeAction);
+                    if (this.OnBlockAllocated == null)
+                        throw new OrleansException("Eviction strategy's OnBlockAllocated is not set for current data adapter, this will affect cache purging");
```

Comment:
this is definitely a valid concern. But my concern is, onBlockAllocated only got set once, if messed up there, all newly allocated buffer will lose the hook to return to the pool. So the memory will be eaten up eventually. Basically fail early vs fail later. 

Maybe a little explain on why queue shifting will affect eviction strategy set up will help. eviction strategy is a dependency of cache creation. EHQueueCache take a IEvicationStrategy param in constructor. In what scenario will eviction strategy exists without the cache?